apps/web/src/components/player/PlayerProfile.tsx

```
@@ -140,17 +245,46 @@ export function PlayerProfile({ initialData, id }: PlayerProfileProps) {
+    return (-b + Math.sqrt(discriminant)) / (2 * a);
+  };
+
+  const getXpForLevel = (lv: number) => {
+    return 127.62 * Math.pow(lv, 2) - 2164.2 * lv + 14553;
+  };
+
   const playtimeSeconds =
     player?.stats?.playtimeSeconds ?? player?.stats?.matchTimeTotal ?? 0;
-  const weaponStats = player?.stats?.weaponStats || [];
+
+  const weaponStats = useMemo(
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)